src/content/docs/docs/guides/vue.mdx

```diff
@@ -0,0 +1,755 @@
+  </form>
+  ...
+</template>
+```
+
+In these changes, you hook up the `onSubmit` handler to the form submit event and bind the `todoText` reactive variable you just created to the input.
+
+You have now implemented the functionality to add new todos, but they still need to be displayed.
+
+<Aside>
+  To verify todos are being created in the database and added to the local
+  state, you can use the [Vue Devtools browser
+  extension](https://devtools.vuejs.org/) and view the `todos` local state.
+  You'll see todos added with the ID returned from the database.
+</Aside>
+
+In the next step, you will display these todos in your app.
+
+## Step 8 — Displaying Todos
+
+In this step, you will fetch and display todos from the collection.
+
+First, update the `doLoginActions` function to fetch todos:
+
+```vue title="App.vue" ins={6}
+<script setup lang="ts">
+...
+
+async function doLoginActions() {
+ isLoggedIn.value = true;
+ todosCollection.getAll().then((docs) => (todos.value = docs));
+}
+
+...
+</script>
+```
+
+In this code, `todosCollection.getAll()` fetches all the todos from the collection, and the results are stored in the `todos` reactive variable.
+
+Next, update the template to display the todos:
+
+```vue title="App.vue" ins={6-12}
+<template>
+  ...
+
+  <form class="todo-form" @submit.prevent="onSubmit">...</form>
+
+  <div class="todos">
+    <div class="card" v-for="todo in todos" :key="todo.id">
+      <div class="todo-text">
+        {{ todo.text }}
+      </div>
+    </div>
+  </div>
+
+  ...
+</template>
+```
+
+Add a few more todos in your browser and notice they get added to the bottom of your list. Usually you want the newest todos to appear first. To do that, you'll reverse the fetched todos.
+
+```vue title="App.vue" del={5,14,22} ins={3,6,7,15,23} mark=", computed"
+<script setup lang="ts">
+...
+import { ref, computed, type Ref } from "vue";
+
+const todos: Ref<Todo[]> = ref([]);
+const fetchedTodos: Ref<Todo[]> = ref([]);
+const todos = computed(() => [...fetchedTodos.value].reverse());
+
+const todosCollection = bzr.collection<Todo>("todos");
+
+async function onSubmit() {
+ const unpersistedTodo = { text: todoText.value, completed: false };
+ const id = await todosCollection.insertOne(unpersistedTodo);
+ todos.value.push({ id, ...unpersistedTodo });
+ fetchedTodos.value.push({ id, ...unpersistedTodo });
+
+ todoText.value = "";
+}
+
+async function doLoginActions() {
+ isLoggedIn.value = true;
+ todosCollection.getAll().then((docs) => (todos.value = docs));
+ todosCollection.getAll().then((docs) => (fetchedTodos.value = docs));
+}
+
+...
+</script>
+```
+
+In these changes, you store fetched todos in a new reactive variable, `fetchedTodos`, and update `todos` to be a computed property that will updated when `fetchedTodos` changes. The spread operator (`...`) creates a new array, as the `reverse` method updates an array in place. Doing `fetchedTodos.value.reverse()` would cause an infinite re-render.
```

Comment:
This will soon be history: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toReversed

Reply:
It's well supported enough. I'll update.

Reply:
Using it actually throws a build error. I'd need to change a compiler option. I'll leave it as is.